include/roboteam_utils/FileLogger.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <string>
+#include <memory>
+#include <mutex>
+
+namespace rtt {
+
+/* A class that makes logging to a file easier and safer.
+ * It handles opening and closing, writing and flushing.
+ * Will append to previously written text in the file.
+ * Throws FailedToOpenFileException if for example the
+ * given folder path of the file does not exist. */
+class FileLogger {
+public:
+    // Will create and open the file you specify. Eg. "log/LOG.txt"
+    explicit FileLogger(const std::string& filePath);
+    // Will close the file
+    ~FileLogger();
+
+    // Writes the line and appends with a newline character
+    void writeNewLine(const std::string& line) const;
+    // Writes the given text to the file
+    void write(const std::string& text) const;
+    // Synchronizes the file with the internal buffer. Read: saves
+    bool flush() const;
+
+private:
+    std::unique_ptr<std::ofstream> stream;
+    mutable std::mutex streamMutex; // Guards the stream
```

Comment:
This is like the only use case for the mutable keyword, but I am wondering if it really makes sense to have all methods as 'const' here. In particular, the internal state of the logger changes on flush() (a second flush may return false?), so I do not think it makes sense to have that function as const.

Reply:
So, the reason why I wanted to have these functions const is so that any other const function could log to a file. Indeed, by making this mutex mutable, I can achieve this, but now I think of it, it might just be better to have all these functions not const, as they 'change' the stream. To still be able to log from const functions, I could make the whole FileLogger member variable a mutable object!